src/js/components/CreatureAbility.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+export default class CreatureAbility extends React.Component {
+
+  constructor(props) {
+    super(props);
+    
+    this.handleValueChanged = this.handleValueChanged.bind(this);
+  }
+
+  handleValueChanged(key, event) {
+    this.props.onChange(key, event.target.value);
+  }
+
+  renderDisplay() {
+    return (
+      <div className='creatureAbility'>
+        <span className='abilityName'>{this.props.name}.</span>
+        <span className='abilityDescription'>{this.props.description}</span>
+      </div>
+    );
+  }
+
+  renderEdit() {
+    return (
+      <div className='creatureAbility'>
+        <input
+          type="text"
+          className="abilityName"
+          value={this.props.name}
+          placeholder="Ability Name"
+          onChange={(event) => this.handleValueChanged('name', event)}
```

Comment:
define `handleValueChanged` as an arrow function so that you can
- avoid `.bind(this)` from Line 8
- also avoid defining inlining functions this would construct a new function on every render
try defining the function and passing it
```
handleNameChanged(event) => {
 this.handleValueChanged('name', event)}
}

onChange={this.handleNameChanged}
```
this would avoid the creation of new functions every time. Theres a lot of instances similar to this here.